server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -61,18 +66,31 @@ export class TenantManager {
      * Validates a tenant's API token
      */
     public async validateToken(tenantId: string, token: string, includeDisabledTenant = false): Promise<void> {
-        const tenantKey = await this.getTenantKey(tenantId, includeDisabledTenant);
-
-        return new Promise<void>((resolve, reject) => {
-            jwt.verify(token, tenantKey, (error) => {
-                if (error) {
-                    // When `exp` claim exists in token claims, jsonwebtoken verifies token expiration.
-                    reject(error instanceof jwt.TokenExpiredError
-                        ? new NetworkError(401, "Token expired.")
-                        : new NetworkError(403, "Invalid token."));
-                } else {
-                    resolve();
+        const tenantKeys = await this.getTenantKeys(tenantId, includeDisabledTenant);
+
+        return jwt.verify(token, tenantKeys.key1, (error1) => {
+            if (!error1) {
+                return Promise.resolve();
+            }
+
+            // if the tenant doesn't have key2, it will be empty string
+            // we should fail token generated with empty string as key
+            if (!tenantKeys.key2) {
+                return Promise.reject(error1 instanceof jwt.TokenExpiredError
+                    ? new NetworkError(401, "Token expired.")
```

Comment:
Nit: Would it be useful to say token1 expired/invalid here and in line 92 say Both tokens failed. While debugging it might be useful to know if we are hitting line 79 or 90. Also, it might be useful to send back if each of the tokens have expired or are invalid in line 90.

Reply:
Makes sense. I will change in the next update. To clarify, here we only have 1 token, but we try to validate the token with 2 keys, if either of the 2 keys can validate the token, the token is validated. And if "exp" claim in the token is verified as expired, then the token is expired. But on line 92 and line 93 we can say it's validated with 2 keys, but both of them failed.